index.ts

```diff
@@ -1,101 +1,38 @@
-import { program } from 'commander'
-import { Wallet } from '@ethersproject/wallet'
-import { JsonRpcProvider, TransactionReceipt } from '@ethersproject/providers'
+import { Wallet, Provider } from 'zksync-web3'
+import { TransactionResponse } from '@ethersproject/providers'
 import { AddressZero } from '@ethersproject/constants'
 import { getAddress } from '@ethersproject/address'
 import fs from 'fs'
 import deploy from './src/deploy'
 import { MigrationState } from './src/migrations'
 import { asciiStringToBytes32 } from './src/util/asciiStringToBytes32'
-import { version } from './package.json'
-
-program
-  .requiredOption('-pk, --private-key <string>', 'Private key used to deploy all contracts')
-  .requiredOption('-j, --json-rpc <url>', 'JSON RPC URL where the program should be deployed')
-  .requiredOption('-w9, --weth9-address <address>', 'Address of the WETH9 contract on this chain')
-  .requiredOption('-ncl, --native-currency-label <string>', 'Native currency label, e.g. ETH')
-  .requiredOption(
-    '-o, --owner-address <address>',
-    'Contract address that will own the deployed artifacts after the script runs'
-  )
-  .option('-s, --state <path>', 'Path to the JSON file containing the migrations state (optional)', './state.json')
-  .option('-v2, --v2-core-factory-address <address>', 'The V2 core factory address used in the swap router (optional)')
-  .option('-g, --gas-price <number>', 'The gas price to pay in GWEI for each transaction (optional)')
-  .option('-c, --confirmations <number>', 'How many confirmations to wait for after each transaction (optional)', '2')
-
-program.name('npx @uniswap/deploy-v3').version(version).parse(process.argv)
-
-if (!/^0x[a-zA-Z0-9]{64}$/.test(program.privateKey)) {
-  console.error('Invalid private key!')
-  process.exit(1)
-}
+const hre = require("hardhat")
 
-let url: URL
-try {
-  url = new URL(program.jsonRpc)
-} catch (error) {
-  console.error('Invalid JSON RPC URL', (error as Error).message)
-  process.exit(1)
-}
+// TODO: think how to pass args
```

Comment:
There is a hardhat task, that you could use to provide ergs. Please do not hardcode pk and url